src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -627,6 +623,19 @@ private static Expression BindParameterFromRouteValueOrQueryString(ParameterInfo
             return BindParameterFromValue(parameter, Expression.Coalesce(routeValue, queryValue), factoryContext);
         }
 
+        private static Expression BindParameterFromBody(Type parameterType, bool allowEmpty, FactoryContext factoryContext)
+        {
+            if (factoryContext.JsonRequestBodyType is not null)
```

Comment:
Does this throw in this case:

```C#
product Handle(Product p1, Product p2)
{
    return p1;
}

record Product(string Name);
```

Reply:
Yep.

https://github.com/dotnet/aspnetcore/pull/31658/files#diff-35b047d1397270b3a7dbbed3bab1d0c5dd76359a51d428a143d3ad07980845e6R717